3711/1/parser/git_hub.rb

```diff
@@ -0,0 +1,78 @@
+require_relative '../ruby_gem'
+require 'json'
+require 'net/http'
+require 'nokogiri'
+require 'open-uri'
+
+module Parser
+  # :reek:TooManyInstanceVariables
+  class GitHub
+    def initialize(gem_name, source_url)
+      @url = source_url
+      @repo = @url[%r{\/[^.:\/]+\/[^.:\/]+}]
+      @gem_data = { 'name' => gem_name, 'url' => @url }
+      @client_id = '26b9a6fc285e836c9f7a'
+      @client_secret = '7ca0b9c0e86047afbac0951d3d505bf7801a5603'
+      @auth_params = "?client_id=#{@client_id}&client_secret=#{@client_secret}"
+    end
+
+    def parse
+      # Collect watched_by, stars and forks
+      collect_watchers_stars_and_forks
+
+      # Collect used_by, contributors and issues
+      collect_used_by_contributors_and_issues
+
+      # Create RubyGem object and return it
+      RubyGem.new(@gem_data)
+    end
+
+    private
+
+    # :reek:TooManyStatements
+    def collect_watchers_stars_and_forks
+      uri = URI("https://api.github.com/repos#{@repo}#{@auth_params}")
+      response = Net::HTTP.get(uri)
+      gem_repo_data = JSON.parse(response)
+      @gem_data['watched_by'] = gem_repo_data['watchers'].to_i
+      @gem_data['stars']      = gem_repo_data['subscribers_count'].to_i
+      @gem_data['forks']      = gem_repo_data['forks'].to_i
+    end
+
+    def collect_used_by_contributors_and_issues
+      collect_used_by
+      collect_contributors
+      collect_issues
+    end
+
+    # rubocop:disable Security/Open
+    def collect_used_by
+      url = "https://github.com#{@repo}/network/dependents"
```

Comment:
You have couple places with similar logic of initialisation of url and opening it.
It should be moved to separate method

Reply:
Done